docs/adr/0004-translations.md

```diff
@@ -0,0 +1,63 @@
+
+[Rails Internationalization](https://guides.rubyonrails.org/i18n.html) (also referred to as 'translations') are a way to extract all of the strings in the application to a central location in `config/locales/en`. The strings can be organized in a hierarchy over one or more files, as below, where we can refer to the reviewed title by writing `I18n.t("publish_document.published.reviewed.title")`.
+
+```
+# publish_document/published.yml
+en:
+  publish_document:
+    published:
+      reviewed:
+        title: Content has been published
+        body: |
+          ‘%{title}’ has been published on GOV.UK.
+
+          It may take 5 minutes to appear live.
+
+```
+
+Rails translations have a few special behaviours, such as pluralization, raw HTML, and variables. The `%{title}` string in the above is an example of a variable, which a developer will set to the title of the document being published.
+
+## Testing
+
+We considered a few approaches for how to extract the strings in our tests. As we started to extract more strings, we had to consider the affect on the readability of the tests and how we could maintain readability/consistency/flexibility.
+
+   * Extract all of the strings in the application and tests. This is consistent and give maximum flexibility for people making changes, but would make the tests hard to read due to the volume of calls to the translation/model APIs.
+   * Extract all of the strings in the application but leave some in the tests. This would be inconsistent for developers to implement and for anyone trying to make a change, as it it's unclear if the tests would need to be fixed. It has the upside that we could select which translations to use in the tests to maintain readability.
+   * Extract a subset of strings in the application and extract the same subset in the tests. This is consistent and flexible, provided the subset is clear and covers all/most of the strings that people want to change.
+
+After trying the first option, we found that extracting links and button labels had the greatest impact on readability; we also think these kinds of strings are unlikely to change. As the original intention behind this work was to enable non-developers to seamlessly edit content, we adopted the third option, which appears to be working well for people making changes, but has caused some confusion among the developers as the subset of strings to extract wasn't specified as it is below.
+
+## Decision
+
+Although we could use translations to extract all of the strings in the application, in some cases we felt this wasn't necessary, or that a different method should be used. The following is a summary of the rules we currently use.
+
+   * **Link and button labels** are not extracted. We think link and button labels are unlikely to change, and extracting them made the application tests harder to read by obfuscating some of the crucial steps in the test with translation keys.
```

Comment:
The approach of not translating clickable things still bugs me. For two reasons:

1. Having a bit translated and other bits untranslated looks to me like I forgot to put something in the locale files:
  
  https://github.com/alphagov/content-publisher/blob/52eae1b87cc7f258111ecacf83536d20c3dece5c/app/views/documents/show/_actions.html.erb#L4-L8

2. I definitely recognise the readability concern in the tests. But what I don't understand is that we do accept this readability drop when it comes to other translated strings. If we were to follow our logic, why are we still translating those strings?

https://github.com/alphagov/content-publisher/blob/52eae1b87cc7f258111ecacf83536d20c3dece5c/spec/features/choose_a_format_spec.rb#L12-L33 

Options in order of my preference:

1. Translate everything consistently
2. Translate the buttons too, but keep the raw strings in the tests
3. Somebody bribe me with cookies so I'll change my mind

Reply:
I've put up a PR for an alternative solution here: https://github.com/alphagov/content-publisher/pull/254/files. Perhaps that could satisfy all parties involved?

Reply:
I'm actually happy with translating everything, and I think @kevindew didn't mind much either way. At the time @davbo was quite concerned about it, though - maybe we should revisit.

@tijmenb your PR reminds me of a concept I once used called [Page Objects](https://robots.thoughtbot.com/better-acceptance-tests-with-page-objects). I think we should revisit using translations directly in tests first, but if it's still a concern, perhaps we could look at doing something like this.